/*
 * ProfaneDB - A Protocol Buffer database.
 * Copyright (C) 2017  "Giorgio Azzinnaro" <giorgio.azzinnaro@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

syntax = "proto3";

package profanedb.protobuf;

option go_package = "gitlab.com/profanedb/protobuf/db";
option csharp_namespace = "ProfaneDB.Protobuf";
option java_package = "com.profanedb.protobuf";
option objc_class_prefix = "PDB";

import "google/protobuf/any.proto";

service Db {
  rpc Get (GetReq) returns (GetResp);

  rpc Put (PutReq) returns (PutResp);

  rpc Delete (DelReq) returns (DelResp);
}

message GetReq {
}

message GetResp {
}

message PutReq {
  google.protobuf.Any serializable = 1;
}

message PutResp {
}

message DelReq {
}

message DelResp {
}
